### DISCUSSION TITLE
Compatibility with OAI UE

### DISCUSSION THREAD

**Discussion by HaoxinSEU** [srsran/srsRAN_Project#395] [General Help] [answered]
Hi community,

We successfully connected srsUE to srsRAN before, and now we are trying to connect [OAI UE](https://gitlab.eurecom.fr/oai/openairinterface5g/-/blob/develop/doc/NR_SA_Tutorial_OAI_nrUE.md?ref_type=heads#5-oai-ue) over the air, since it supports TDD. However, the OAI UE cannot connect and keeps outputting:
```
[NR_PHY]   Starting sync detection
[PHY]   [UE thread Synch] Running Initial Synch 
[PHY]   [UE] nr_synchro_time: Sync source = 1, Peak found at pos 281320, val = 4411826 (66 dB) avg 58 dB, ffo 0.158434
[PHY]   PSS execution duration 41333 microseconds 
[PHY]   [UE0] Initial sync : Estimated PSS position 281320, Nid2 1
[PHY]   sync_pos 281320 ssb_offset 281212 
[PHY]   Calling sss detection (normal CP)
[PHY]   TDD Normal prefix: CellId 277 metric 976, phase 15, pbch 1
[PHY]   [UE] nr_synchro_time: Sync source = 0, Peak found at pos 143976, val = 329621905 (85 dB) avg 60 dB, ffo -0.330922
[PHY]   PSS execution duration 41055 microseconds 
[PHY]   [UE0] Initial sync : Estimated PSS position 143976, Nid2 0
[PHY]   sync_pos 143976 ssb_offset 143868 
[PHY]   Calling sss detection (normal CP)
[PHY]   TDD Normal prefix: CellId 405 metric 1143, phase 3, pbch 1
[PHY]   [UE0] Initial sync : Estimated power: 0 dB
[PHY]   [SCHED][UE] Check absolute frequency DL 3604320000.000000, UL 3604320000.000000 (RF card 0, oai_exit 0, channel 0, rx_num_channels 1)

```

Since the connection is over the air, we expect that OAI UE would behave like a normal COTS modem, and connect.

We are wondering if srsRAN is compatible with OAI UE? And if anyone managed to connect OAI UE to srsRAN?

Many thanks for the help!

**Comment by brendan-mcauliffe** [comment, marked as answer]
Hi @HaoxinSEU, 

In theory, you can connect OAI UE to the srsRAN CU/DU. However, we have not tested this in-house. If the same configuration for the CU/DU is working with a COTS UE or srsUE, then this is an OAI UE issue and I suggest you look for help via their forums.

**Reply by HaoxinSEU** [reply]
Hi @brendan-mcauliffe ,

Thanks for your reply! Yes indeed, we find some discussion about the OAI UE from 2 months ago, which said that it cannot work with commercial gNB. So I suppose that's the reason.

Still it would be very nice if the srsUE can work in TDD mode in the future, as in general TDD is used more in real scenarios, lol

Thanks again!

**Comment by pengnist** [comment]
Hi @HaoxinSEU, we are thinking about doing the same thing. I am wondering if you worked further on the OAI UE <-> srsRAN gNB connection after this thread. Thanks!

**Reply by HaoxinSEU** [reply]
Hi, we don't go further into this. But we believe it's an interesting topic to investigate...

**Comment by Moussa-Guemdani** [comment]
Hello @HaoxinSEU , µI am trying the same thing
did you manage to connect OAI nrUE with srsRAN gNB?

Thank you.

**Comment by Alan1321** [comment]
@Moussa-Guemdani Were you able to connect? I am also testing the same thing.
@HaoxinSEU Could you please add your configuration files for srsRAN gNB, and OAI nrUE.
